azurerm/resource_arm_automation_dsc_nodeconfiguration_test.go

```diff
@@ -0,0 +1,160 @@
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMAutomationDscNodeConfigurationDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMAutomationDscNodeConfiguration_basic(ri, testLocation()),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMAutomationDscNodeConfigurationExists(resourceName),
+				),
+			},
+			{
+				ResourceName:            resourceName,
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"content"},
+			},
+		},
+	})
+}
+
+func testCheckAzureRMAutomationDscNodeConfigurationDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*ArmClient).automationDscNodeConfigurationClient
+	ctx := testAccProvider.Meta().(*ArmClient).StopContext
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "azurerm_automation_dsc_nodeconfiguration" {
+			continue
+		}
+
+		name := rs.Primary.Attributes["name"]
+		accName := rs.Primary.Attributes["account_name"]
+
+		resourceGroup, hasResourceGroup := rs.Primary.Attributes["resource_group_name"]
+		if !hasResourceGroup {
+			return fmt.Errorf("Bad: no resource group found in state for Automation Dsc Node Configuration: '%s'", name)
+		}
+
+		resp, err := conn.Get(ctx, resourceGroup, accName, name)
+
+		if err != nil {
+			if utils.ResponseWasNotFound(resp.Response) {
+				return nil
+			}
+
+			return err
+		}
+
+		return fmt.Errorf("Automation Dsc Node Configuration still exists:\n%#v", resp)
+
+	}
+
+	return nil
+}
+
+func testCheckAzureRMAutomationDscNodeConfigurationExists(name string) resource.TestCheckFunc {
+
+	return func(s *terraform.State) error {
+		// Ensure we have enough information in state to look up in API
+		rs, ok := s.RootModule().Resources[name]
+		if !ok {
+			return fmt.Errorf("Not found: %s", name)
+		}
+
+		name := rs.Primary.Attributes["name"]
+		accName := rs.Primary.Attributes["account_name"]
+
+		resourceGroup, hasResourceGroup := rs.Primary.Attributes["resource_group_name"]
+		if !hasResourceGroup {
+			return fmt.Errorf("Bad: no resource group found in state for Automation Dsc Node Configuration: '%s'", name)
+		}
+
+		conn := testAccProvider.Meta().(*ArmClient).automationDscNodeConfigurationClient
+		ctx := testAccProvider.Meta().(*ArmClient).StopContext
+
+		resp, err := conn.Get(ctx, resourceGroup, accName, name)
+
+		if err != nil {
+			if utils.ResponseWasNotFound(resp.Response) {
+				return fmt.Errorf("Automation Dsc Node Configuration '%s' (resource group: '%s') does not exist", name, resourceGroup)
+			}
+
+			return fmt.Errorf("Bad: Get on automationDscNodeConfigurationClient: %s\nName: %s, Account name: %s, Resource group: %s OBJECT: %+v", err, name, accName, resourceGroup, rs.Primary)
+		}
+
+		return nil
+	}
+}
+
+func testAccAzureRMAutomationDscNodeConfiguration_basic(rInt int, location string) string {
+	return fmt.Sprintf(`
+resource "azurerm_resource_group" "test" {
```

Comment:
This terraform should be properly formatted, see my comments above

Reply:
done